### PR TITLE
PIMS-1474: Save Evaluations and Fiscals alongside Parcels

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

  
[PIMS-1474 ](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-1474)

* Fixed various issues with entity definitions for parcels, parcel fiscals, and parcel evaluations.
* Parcel evaluation will use Year (int) instead of Date (timestamp)
* The service for getting a parcel by id now returns its child relations as well.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: express-api/src/typeorm/Entities/abstractEntities/Property.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export abstract class Property extends BaseEntity {
   @Column({
     type: 'point',
     transformer: {
-      to: (obj: GeoPoint) => `${obj.x},${obj.y}`, // Going to the database. Format into point.
+      to: (obj: GeoPoint) => `${obj?.x},${obj.y}`, // Going to the database. Format into point.
```
Comment (dbarkowsky) [inline express-api/src/typeorm/Entities/abstractEntities/Property.ts:60]: Why the ? and why only on the first one?

File: express-api/src/services/parcels/parcelServices.ts [hunk 2]
```diff
@@ -90,10 +90,8 @@ const updateParcel = async (incomingParcel: DeepPartial<Parcel>) => {
   if (findParcel == null || findParcel.Id !== incomingParcel.Id) {
     throw new ErrorWithCode('Parcel not found', 404);
   }
-  await parcelRepo.update({ Id: findParcel.Id }, incomingParcel);
-  // update function doesn't return data on the row changed. Have to get the changed row again
-  const newParcel = await getParcelById(incomingParcel.Id);
-  return newParcel;
+
+  return parcelRepo.save(incomingParcel);
```
Comment (dbarkowsky) [inline express-api/src/services/parcels/parcelServices.ts:94]:
```suggestion
  return await parcelRepo.save(incomingParcel);
```
This feels safer if we await the promise.

File: express-api/src/services/parcels/parcelServices.ts [hunk 3]
```diff
@@ -113,7 +111,18 @@ const getParcelByPid = async (parcelPid: number) => {
  * @returns     findParcel Parcel data matching ID passed in.
  */
 const getParcelById = async (parcelId: number) => {
-  return parcelRepo.findOne({ where: { Id: parcelId } });
+  return parcelRepo.findOne({
```
Comment (dbarkowsky) [inline express-api/src/services/parcels/parcelServices.ts:114]:
```suggestion
  return await parcelRepo.findOne({
```
Same await concern

**Comment by dbarkowsky** [comment]
I assume we'll need this same setup for the Buildings + their evaluations and fiscals. Same ticket or separate?

File: express-api/src/services/parcels/parcelServices.ts (hunk 2 above)
Comment (GrahamS-Quartech) [reply express-api/src/services/parcels/parcelServices.ts:94]: The caller will need to await the returned Promise of this function either way.

File: express-api/src/typeorm/Entities/abstractEntities/Property.ts (hunk 1 above)
Comment (GrahamS-Quartech) [reply express-api/src/typeorm/Entities/abstractEntities/Property.ts:60]: Crtl+Z mistake woops.

**Comment by GrahamS-Quartech** [comment]
> I assume we'll need this same setup for the Buildings + their evaluations and fiscals. Same ticket or separate?
I was gonna say separate, but it is basically the exact same process so... Maybe I'll just work on adding that into this PR too.
> Also, we should probably add a migration for this.
Forgot about this, I will generate them.

**Comment by GrahamS-Quartech** [comment]
I needed to modify the generated migration quite a bit, what TypeORM spat out was kinda weird. Please test running and reverting the migration if you can, though it seems fine through my testing.

**Comment by dbarkowsky** [comment]
> I needed to modify the generated migration quite a bit, what TypeORM spat out was kinda weird. Please test running and reverting the migration if you can, though it seems fine through my testing.

Doesn't seem to run for me at the moment. The composite keys needed to be tweaked for the ParcelEvaluations, ParcelFiscals, BuildingEvaluations, and BuildingFiscals. 
Are you running the migration with data already in the database? I notice it fails for me when it tries to add a constraint because there's apparently more than one record with the same evaluation key id and building id.

**Review by dbarkowsky — APPROVED** [review]
Tested with cleaned up data from prod. Seems to be working well in both directions. 
I added some indexes for the 3-way composite primary key.

**Pull request merged**